pymodbus/payload.py

```diff
@@ -69,15 +68,14 @@ def _pack_words(self, fstring: str, value) -> bytes:
         :return:
         """
         value = pack(f"!{fstring}", value)
-        wordorder = WC.get(fstring.lower()) // 2  # type: ignore[operator]
-        upperbyte = f"!{wordorder}H"
-        payload = unpack(upperbyte, value)
-
-        if self._wordorder == Endian.LITTLE:
-            payload = payload[::-1]
-
-        fstring = self._byteorder + "H"
-        return b"".join(pack(fstring, word) for word in payload)
+        if Endian.LITTLE in {self._byteorder, self._wordorder}:
+            value = array("H", value)
+            if self._byteorder == Endian.LITTLE:
+                value.byteswap()
+            if self._wordorder == Endian.LITTLE:
+                value.reverse()
```

Comment:
this is wrong because it reverses the byte order not the word order

e.g. value = 0123 result should be 2301, NOT 3210